sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3642,6 +3641,10 @@ __ESIMD_API std::enable_if_t<
     simd<T, N>>
 gather(AccessorT acc, OffsetSimdViewT byte_offsets, simd_mask<N / VS> mask,
        simd<T, N> pass_thru, PropertyListT props = {}) {
+  static_assert(N / VS ==
```

Comment:
We might want to make an internal tracker to add these asserts to the APIs we already changed